src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1803,7 +1803,7 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
         }
     }
 
-    fn resolve_elided_lifetimes(&mut self, lifetime_refs: &'tcx [hir::Lifetime]) {
+    fn resolve_elided_lifetimes(&mut self, lifetime_refs: &'tcx [hir::Lifetime], deprecated: bool) {
```

Comment:
@Dylan-DPC 

Here is an example of adding a lint from the code in question:

https://github.com/rust-lang/rust/blob/fdc18b3067b5bad257ccbe7400e3c4fb617e9e18/src/librustc/middle/resolve_lifetime.rs#L1228-L1234

I think you should be able to do something similar from this function here when `deprecated` is true.

Reply:
Hey Thanks
I have followed that code already was wondering if i had to create the `LifetimeContext` object just like what is being done in the function. I guess i have to.

Reply:
I don' think you have to do that, no. I am confused. You should have a `LifetimeContext` available. 

Basically, that function does some things in addition to creating the lint -- that is why it makes the new `LifetimeContext` struct -- you are not doing those things. The only thing you should need is the call to `struct_span_lint_node`

Reply:
> I have followed that code already

What do you mean by "followed" here? I don't see (in the PR) any calls to `struct_span_lint_node`, maybe you forgot to push a commit?

Reply:
You should not have to create a `LifetimeContext`, no. That function does a few other things in addition to issuing the lint -- in particular, it pushes a "subscope" -- and that is why it creates a `LifetimeContext`. You can just use the one you have, I expect.

Reply:
> What do you mean by "followed" here? I don't see (in the PR) any calls to struct_span_lint_node, maybe you forgot to push a commit?

Ah I meant the code you had mentioned earlier. 

My question is: do I have access to this.tcx directly? I thought I didn't